server/src/main/java/org/opensearch/common/time/JavaDateFormatter.java

```diff
@@ -285,29 +290,43 @@ public TemporalAccessor parse(String input) {
      * @throws DateTimeParseException when unable to parse with any parsers
      */
     private TemporalAccessor doParse(String input) {
-        if (parsers.size() > 1) {
-            Object object = null;
-            if (canCacheLastParsedFormatter && lastParsedformatter != null) {
-                ParsePosition pos = new ParsePosition(0);
-                object = lastParsedformatter.toFormat().parseObject(input, pos);
-                if (parsingSucceeded(object, input, pos)) {
-                    return (TemporalAccessor) object;
-                }
+        final ParsePosition pos = new ParsePosition(0);
```

Comment:
Why `if (parsers.size() > 1)` was removed?

Reply:
It was to unify the code logic, in the old implementation, when `parsers.size() < 1` the code logic is `this.parsers.get(0).parse(input);` which may also throw DateTimeException.

Reply:
But this is legit, the `parse(CharSequence)` could throw `DateTimeParseException`.

Reply:
Or you mean that `parse(CharSequence)` throws  `DateTimeException` instead?

Reply:
Just double checked the source code of parse(CharSequence), it catches DateTimeParseException and RuntimeException, re-throw DateTimeParseException in both case, hence it won't throw DateTimeException.